### PR TITLE
ci: lint, security scan, envtest, codecov

### PR DESCRIPTION
**Pull request by psturc**
### JIRA link
https://issues.redhat.com/browse/PLNSRVCE-281

### To do
- [x] Github Actions enabled for this repo (by repo owner)
- [x] [Codecov App](https://github.com/apps/codecov) installed by owner of this repo

You can see [here](https://github.com/psturc/build-service/pull/2) how it looks in a PR when GH Actions and Codecov is enabled

### CODE REVIEW

**Review by gabemontero — APPROVED** [review]

**Comment by Michkov** [comment]
@psturc Github actions set to  `Allow all actions and reusable workflows`, [Codecov App](https://github.com/apps/codecov) is also already installed

File: controllers/suite_util_test.go [hunk 1]
```diff
@@ -293,9 +293,9 @@ func createBuildTaskRunWithImage(resourceKey types.NamespacedName, image string)
 }
 
 func deleteAllTaskRuns() {
-	k8sClient.DeleteAllOf(ctx, &tektonapi.TaskRun{}, &client.DeleteAllOfOptions{
+	Expect(k8sClient.DeleteAllOf(ctx, &tektonapi.TaskRun{}, &client.DeleteAllOfOptions{
```
Comment (mmorhun) [inline controllers/suite_util_test.go:296]: If no objects of the kind is found, the assert is fail.

**Review by mmorhun — APPROVED** [review]
LGTM

File: controllers/suite_util_test.go (hunk 1 above)
Comment (psturc) [reply controllers/suite_util_test.go:296]: 👍  will fix in a follow-up PR

**Pull request merged**